lib/index.js

```diff
@@ -332,6 +336,72 @@ export default {
     return provider
   },
 
+  provideOutlines(): OutlineProvider {
+    const outlineOptions: OutlineOptions = {
+      showKeywords: {
+        export: false,
+        default: false,
+        const: false,
+        var: false,
+        let: false,
+        class: false,
+        function: false,
+        type: false,
+      },
+      showFunctionArgs: false,
+    }
+    this.subscriptions.add(atom.config.observe('flow-ide.outline', (outline) => {
+      outlineOptions.showKeywords.export = outline.showExport
+      outlineOptions.showFunctionArgs = outline.showFunctionArgs
+    }))
+
+    const provider = {
+      name: 'flow-ide',
+      priority: 1,
+      grammarScopes,
+      updateOnEdit: true,
+      getOutline: async (editor: TextEditor) => {
+        const filePath = editor.getPath()
+        if (!filePath) {
+          return null
+        }
+
+        const fileDirectory = Path.dirname(filePath)
+        const configFile = await findCachedAsync(fileDirectory, '.flowconfig')
+        if (!configFile) {
+          return null
+        }
+
+        const flowOptions = ['ast']
+
+        let result
+        try {
+          result = await exec(await this.getExecutablePath(fileDirectory), flowOptions, {
+            cwd: fileDirectory,
+            stdin: editor.getText(),
+            timeout: 60 * 1000,
+            uniqueKey: 'flow-ide-ast',
+            ignoreExitCode: true,
+          })
+          if (result === null) {
```

Comment:
`return null` for not changing the output is only supported by linter, we should return an empty array or whatever the consumer expects as default value

Reply:
Yep, you are right :+1: !